request.cpp

```diff
@@ -0,0 +1,61 @@
+#include <string>
+#include <iostream>
+#include "request.hpp"
+#include <boost/regex.hpp>
+#include <memory>
+
+
+std::unique_ptr<Request> Request::Parse(const std::string& raw_request){
+
+  std::unique_ptr<Request> req = std::unique_ptr<Request>(new Request);
+  req->_raw_request = raw_request;
+  
+  //Set request type, path, method
+  boost::regex first_line_expr{"^([a-zA-Z]+) (\\/[\?a-zA-Z0-9.\\/]*) ([a-zA-Z0-9.\\/]*)"};
+  //boost::regex path_expr{"(\\/[\?a-zA-Z0-9]*)"};
+  boost::smatch what;
+  
+  if (boost::regex_search(raw_request, what, first_line_expr)){
+    if (what.size() < 4)
+      return false; 
+
+    req->_method = what[1];
+    req->_uri = what[2];
+    req->_version = what[3];
+
+      /*
+      boost::regex_search(full_path, what, path_expr);
+      uri = what[0];
+      _filePath = full_path.substr(uri.size(), full_path.size() - _path.size());
+      */
+  }
+
```

Comment:
Need to find the right regex to extract (Header Name: Header Value) and insert each one into _headers private member.